build-tools/packages/build-cli/src/lib/package.ts

```diff
@@ -497,3 +498,40 @@ export function filterVersionsOlderThan(
         return diff <= numBusinessDays;
     });
 }
+
+export function getFluidDependencies(
+    context: Context,
+    releaseGroupOrPackage: ReleaseGroup | ReleasePackage,
+): [releaseGroups: PackageVersionMap, packages: PackageVersionMap] {
+    const releaseGroups: PackageVersionMap = {};
+    const packages: PackageVersionMap = {};
+    let packagesToCheck: Package[];
+
+    if (isReleaseGroup(releaseGroupOrPackage)) {
+        packagesToCheck = context.packagesInReleaseGroup(releaseGroupOrPackage);
+    } else {
+        const independentPackage = context.fullPackageMap.get(releaseGroupOrPackage);
+        assert(
+            independentPackage !== undefined,
+            `Package not found in context: ${releaseGroupOrPackage}`,
+        );
+        packagesToCheck = [independentPackage];
+    }
+
+    for (const p of packagesToCheck) {
+        for (const dep of p.combinedDependencies) {
+            const pkg = context.fullPackageMap.get(dep.name);
+            if (pkg !== undefined) {
+                if (pkg.monoRepo === undefined) {
+                    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                    packages[pkg.name] = semver.minVersion(dep.version)!.version;
```

Comment:
I'm not sure if you have a good "fail" function (with return type `never`) to use here, but I prefer to actually detect and error on undefined/null in cases like this instead of suppressing the linter:
```suggestion
                    packages[pkg.name] = semver.minVersion(dep.version)?.version ?? fail("missing minVersion");
```

We use this pattern in tree, though we currently don't have great integration for it with Fluid's assert message code thing.

Reply:
Fixed in latest iteration.